### ISSUE TITLE
[DatePicker] no aria-label passed down to input element (mobile)

### ISSUE THREAD

**Issue by sodenn** [mui/mui-x#8146]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:
https://codesandbox.io/s/quizzical-dawn-rwqcdn?file=/demo.tsx

Steps:
1. Open the codesandbox on an iOS device
2. Inspect the aria-label of the input element

### Current behavior 😯

In v6.0.0 there is a new `slotProps` that can be used to set the aria-label of the input element:

```jsx
<DatePicker
  label="Basic date picker"
  slotProps={{
    textField: (props) => ({
      inputProps: {
        "aria-label": "Basic date picker",
        ...props.inputProps
      }
    })
  }}
/>
```

This works fine for desktop but not for mobile devices. This is a problem, especially when running automated tests on iOS environments, since the input element can no longer be selected via role + aria-label.

The aria-label always contains "Choose date" although something else is defined:
<img width="1048" alt="Screenshot_2023-03-05_at_21_11_22" src="https://user-images.githubusercontent.com/232054/222989210-d6ef0e58-b32a-4c7f-817f-1072efe1bd35.png">

### Expected behavior 🤔

The aria-label defined in `slotProps` should be passed to the input element as it is already done on desktop devices.

### Your environment 🌎

OS: iOS 16.2
@mui/x-date-pickers: 6.0.0

**Comment by sodenn** [comment]
I accidentally override the aria-label by spreading the object at the wrong position. This works fine:
```jsx
<DatePicker
  label="Basic date picker"
  slotProps={{
    textField: (props) => ({
      inputProps: {
        ...props.inputProps,
        "aria-label": "Basic date picker",
      }
    })
  }}
/>
```
